addons/google_calendar/models/google_calendar.py

```diff
@@ -398,7 +398,17 @@ def update_recurrent_event_exclu(self, instance_id, event_ori_google_id, event_n
         url = "/calendar/v3/calendars/%s/events/%s?access_token=%s" % ('primary', instance_id, self.get_token())
         headers = {'Content-type': 'application/json'}
 
-        data.update(recurringEventId=event_ori_google_id, originalStartTime=event_new.recurrent_id_date, sequence=self.get_sequence(instance_id))
+        _originalStartTime = dict()
+        if event_new.allday:
+            _originalStartTime['date'] = event_new.recurrent_id_date.strftime("%Y-%m-%d")
+        else:
+            _originalStartTime['datetime'] = event_new.recurrent_id_date.strftime("%Y-%m-%dT%H:%M:%S.%fz")
+
+        data.update(
+            recurringEventId=event_ori_google_id,
+            originalStartTime=_originalStartTime,
+            sequence=self.get_sequence(instance_id)
+        )
```

Comment:
Hello @agr-odoo ,

I'm troubleshooting a problem where the new originalStartTime format introduced here is causing an issue on a couple of databases that run on Odoo.sh.

I keep on getting errors in the logs when the cron is executing on a database.

`<time> <worker_id> ERROR  <db> odoo.addons.google_account.models.google_service: Bad google request : b'{\n "error": {\n  "errors": [\n   {\n    "domain": "global",\n    "reason": "invalid",\n    "message": "Invalid Value"\n   }\n  ],\n  "code": 400,\n  "message": "Invalid Value"\n }\n}\n' ! `

I managed to extract the call and replay it to reproduce the error.

Funny thing is that when I change the  

`"originalStartTime": {"datetime": "2019-11-12T16:00:00.000000z"},`
by
`"originalStartTime": "2019-11-15T16:00:00.000+00:00",`

in my request's data I get a 200 OK response.

I'm a bit curious where you read 

> "Moreover the api usage was wrong as 'originalStartTime' is
> not a date but a complex object holding the date."

As in the current documentation that I found, it still looks like it should be a regular string.

https://developers.google.com/calendar/recurringevents

Was this a necessary change during your tests? If not I'll make a PR reverting this change. Also, has a change been made in v11.0 too or is this the earliest version?

Reply:
Well, I can reproduce your issue when the recursive event is not an all-day event.
I see the doc reference you are pointing at and I reply with https://developers.google.com/calendar/v3/reference/events#originalStartTime
so we can conclude that either someone forgot to update one of the two pages or both of them are valid requests.

I started investigating "manually", when sending requests to google if:
* originalStartTime is dict
  * all day event -> OK
  * normal event -> Error
* originalStartTime is string
  * all day event -> OK
  * normal event -> OK

Note that if you inspect a reply from GC in any case you will see something like
'originalStartTime': {'dateTime': '2019-11-08T09:00:00+01:00', 'timeZone': 'Europe/Brussels'}

After an hour and all this considerations I noticed a typo in **dateTime**, here it is the PR:
https://github.com/odoo/odoo/pull/40595